sklearn/linear_model/_base.py

```diff
@@ -419,6 +420,13 @@ class LinearRegression(MultiOutputMixin, RegressorMixin, LinearModel):
         ``-1`` means using all processors. See :term:`Glossary <n_jobs>`
         for more details.
 
+    solver : str, default="lstsq"
+        The solver to use. ``"lstsq"`` uses a SVD-based least-squares
+        approach, by calling ``scipy.linalg.lstsq``. ``"cholesky"`` uses the
+        Cholesky decomposition. If X is singular, then ``"cholesky"`` will
+        instead use an SVD-based solver. ``"cholesky"`` does not support `X`
+        matrices which are both singular and sparse.
```

Comment:
I see this is the behavior for `Ridge`, but this is still really strange behavior.